spec/api-spec.js

```diff
@@ -123,38 +123,35 @@ describe('Get Requests', () => {
       }
     }
     server.post(reqPost, res)
-      .then((err) => {
-        if (err) {
-          done.fail(err)
-        } else {
-          expect(res.status).toHaveBeenCalledWith(200)
-          server.get(reqGet, res)
-            .then((err) => {
-              if (err) {
-                done.fail(err)
-              } else {
-                expect(res.status).toHaveBeenCalledWith(200)
-                expect(res.format).toHaveBeenCalledWith({
-                  json: jasmine.any(Function)
-                })
-                expect(redisResponse).toEqual('acompletelynewandoriginaltoken')
-                done()
-              }
+      .then((result) => {
+        expect(res.status).toHaveBeenCalledWith(200)
+        server.get(reqGet, res)
+          .then((result) => {
+            expect(res.status).toHaveBeenCalledWith(200)
+            expect(res.format).toHaveBeenCalledWith({
+              json: jasmine.any(Function)
             })
-        }
+            expect(result).toEqual(reqPost.body.developer.token)
+            done()
+          }).catch((rej) => {
+            done.fail(rej)
+          })
       })
   })
 
   it('should indicate that a username does not exist', (done) => {
     const req = {query: {username: 'testrandomperson', token: 'acompletelynewandoriginaltoken'}}
     server.get(req, res)
-      .then((err) => {
-        expect(err).not.toBe(null)
-        expect(res.status).toHaveBeenCalledWith(500)
+      .then((result) => {
+        done.fail(result)
+      })
+      .catch((errResult) => {
+        expect(errResult).not.toBe(null)
+        expect(res.status).toHaveBeenCalledWith(520)
```

Comment:
Again with the `520` error-code. This is a Bad Idea. No need to invent your own error-codes here. This is a simple `403` condition. Use that error-code instead.